src/Sulu/Bundle/WebsiteBundle/Controller/DefaultController.php

```diff
@@ -64,4 +62,38 @@ public function redirectAction(Request $request)
     {
         return new RedirectResponse($request->get('url'), 301);
     }
+
+    /**
+     * Resolve the redirect URL, appending any additional path data
+     *
+     * @param string $url Original webspace URI
+     * @param string $redirectUrl  Redirect webspace URI
+     * @param string $requestUri The actual incoming request URI
+     *
+     * @return string URL to redirect to
+     */
+    protected function resolveRedirectUrl($portalUrl, $redirectUrl, $requestUri)
```

Comment:
Should probably be private as this is basically evil and should be somewhere else.